modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/ShadedReliefNode.java

```diff
@@ -0,0 +1,317 @@
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.renderer.lite.gridcoverage2d;
+
+import it.geosolutions.jaiext.range.NoDataContainer;
+import it.geosolutions.jaiext.range.Range;
+import it.geosolutions.jaiext.shadedrelief.*;
+import java.awt.geom.AffineTransform;
+import java.awt.image.RenderedImage;
+import java.awt.image.renderable.RenderedImageFactory;
+import java.util.HashMap;
+import java.util.List;
+import javax.measure.Unit;
+import javax.media.jai.JAI;
+import javax.media.jai.OperationDescriptor;
+import javax.media.jai.OperationRegistry;
+import javax.media.jai.ROI;
+import javax.media.jai.RenderedOp;
+import javax.media.jai.registry.RenderedRegistryMode;
+import org.geotools.coverage.GridSampleDimension;
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.geotools.coverage.grid.GridCoverageFactory;
+import org.geotools.coverage.grid.GridGeometry2D;
+import org.geotools.factory.Hints;
+import org.geotools.image.ImageWorker;
+import org.geotools.renderer.i18n.ErrorKeys;
+import org.geotools.renderer.i18n.Errors;
+import org.geotools.resources.coverage.CoverageUtilities;
+import org.geotools.styling.ShadedRelief;
+import org.geotools.styling.StyleVisitor;
+import org.geotools.util.SimpleInternationalString;
+import org.opengis.coverage.grid.GridCoverage;
+import org.opengis.filter.expression.Expression;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.referencing.operation.MathTransform;
+import org.opengis.util.InternationalString;
+import si.uom.SI;
+
+/**
+ * This implementations of {@link CoverageProcessingNode} takes care of the {@link ShadedRelief}
+ * element of the SLD 1.0 spec.
+ *
+ * @author Daniele Romagnoli, GeoSolutions SAS
+ */
+class ShadedReliefNode extends StyleVisitorCoverageProcessingNodeAdapter
+        implements StyleVisitor, CoverageProcessingNode {
+
+    static {
+        // Initialize JAIExt operations
+        OperationRegistry registry = JAI.getDefaultInstance().getOperationRegistry();
+        OperationDescriptor op = new ShadedReliefDescriptor();
+
+        // registering the descriptor
+        registry.registerDescriptor(op);
+        String descName = op.getName();
+
+        // registering the RIF
+        RenderedImageFactory rif = new ShadedReliefRIF();
+        registry.registerFactory(
+                RenderedRegistryMode.MODE_NAME, descName, "org.geotools.gce.processing", rif);
+    }
+
+    private boolean brigthnessOnly = false;
```

Comment:
Probably should be `brightnessOnly`

Reply:
Yep, a typo, anyway it's private and won't be leaked outside. Thanks for spotting it!